### PR TITLE
Merge DFL kernel latest patches

### PR DESCRIPTION
**Pull request by Muralix**

### CODE REVIEW

File: drivers/tty/serial/8250/8250_dfl.c [hunk 1]
```diff
@@ -2,160 +2,176 @@
...
+	u64 fifo_len, clk_freq, reg_layout;
+	u32 reg_width;
+	int ret;
 
-	dfluart->uart_clk = readq(param_base + off + DFHv1_PARAM_DATA);
-	dev_dbg(dfluart->dev, "UART_CLK_ID %llu Hz\n", dfluart->uart_clk);
+	ret = dfh_get_u64_param_val(dfl_dev, DFHv1_PARAM_ID_CLK_FRQ, &clk_freq);
+	if (ret) {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 9, 0) && RHEL_RELEASE_CODE < 0x803
+		return dev_err(dev,
```
Comment (rchriste1) [inline drivers/tty/serial/8250/8250_dfl.c:62]: On CentOS 8.2 there is compilation issue:
/home/rc/acbackport/backport/linux-dfl-backport/drivers/tty/serial/8250/8250_dfl.c:62:10: error: void value not ignored as it ought to be
   return dev_err(dev,
          ^~~~~~~~~~~~
This occurs in all the places in this file where you put in dev_err.

File: drivers/mfd/intel-m10-bmc-pmci.c [hunk 2]
```diff
@@ -1,148 +1,242 @@
...
  *
+ * Copyright (C) 2020-2023 Intel Corporation.
  */
 
 #include <linux/bitfield.h>
+#include <linux/device.h>
 #include <linux/dfl.h>
+#include <linux/mfd/core.h>
 #include <linux/mfd/intel-m10-bmc.h>
+#include <linux/minmax.h>
```
Comment (rchriste1) [inline drivers/mfd/intel-m10-bmc-pmci.c:14]: CentOS 8.4 does not have this include file:
  CC [M]  drivers/mfd/intel-m10-bmc-pmci.o
drivers/mfd/intel-m10-bmc-pmci.c:14:10: fatal error: linux/minmax.h: No such file or directory
 #include <linux/minmax.h>
          ^~~~~~~~~~~~~~~~
compilation terminated.

**Review by rchriste1 — COMMENTED** [review]
A simple test with N5010 card on Rocky Linux 9.1 runs OK.

**Comment by Muralix** [comment]
Required more changes to support for older kernels. So going for new pull request after corrections.

**Pull request closed without merging**